google/gax/api_callable.py

```diff
@@ -41,59 +41,63 @@
 _MILLIS_PER_SECOND = 1000
 
 
-def _add_timeout_arg(a_func, timeout):
-    """Updates a_func so that it gets called with the timeout as its final arg.
+class _RetryableWrapper(object):
```

Comment:
@tbetbetbe made a past decision to move away from using callable objects to using our current functional/"decorator-like" style. What is the advantage in using an object here?